starlette/middleware/gzip.py

```diff
@@ -42,6 +42,7 @@ def __init__(self, app: ASGIApp, minimum_size: int, compresslevel: int = 9) -> N
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         self.send = send
         await self.app(scope, receive, self.send_with_gzip)
+        self.gzip_file.close()
```

Comment:
@graingert Is this good enough?

Reply:
You might want to do it in a finally, or even better use `with`

```python
with BytesIO() as buffer, with GzipFile(..., buffer) as gzip_file:
    await self.app(scope, receive, functools.partial(self.send_with_gzip, buffer, gzip_file))
```